File: platforms/ios/runtime/Sources/NativeblocksRuntime/api/NativeblocksEdition.swift
import Foundation

/// Represents the different editions of the Nativeblocks framework.
public enum NativeblocksEdition {
    /// Cloud-based configuration.
    /// - Parameters:
    ///   - endpoint: The API endpoint for fetching frame definitions and other resources.
    ///   - apiKey: The API key for authenticating requests to the server.
    ///   - developmentMode: Indicates if development mode is enabled for testing and debugging purposes.
    case cloud(endpoint: String, apiKey: String, developmentMode: Bool)

    /// Community-based configuration using local frame data.
    /// - Parameter frameData: A dictionary where each key is a route and each value is the corresponding frame URL.
    case community(frameData: [String: String])
}

File: platforms/android/runtime/core/src/main/java/io/nativeblocks/runtime/api/NativeblocksEdition.kt
package io.nativeblocks.runtime.api

/**
 * Represents the different editions of the Nativeblocks framework.
 */
sealed interface NativeblocksEdition {

    /**
     * Cloud-based configuration for the Nativeblocks framework.
     * @param endpoint The API endpoint for fetching frame definitions and other resources.
     * @param apiKey The API key for authenticating requests to the server.
     * @param developmentMode Indicates if development mode is enabled for testing and debugging purposes.
     */
    data class Cloud(
        val endpoint: String,
        val apiKey: String,
        val developmentMode: Boolean
    ) : NativeblocksEdition

    /**
     * Community-based configuration for the Nativeblocks framework.
     * @param framesData A map where each key represents a route, and each value is the URL or path to the corresponding frame data.
     */
    class Community(
        val framesData: Map<String, String>,
    ) : NativeblocksEdition
}
